templates/enmasse-controller-manager/020-ClusterRole-enmasse-controller-manager.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    app: enmasse
+  name: enmasse.io:enmasse-controller-manager
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "configmaps", "secrets", "pods", "services", "persistentvolumeclaims" ]
+    verbs: [ "get", "list", "watch", "update", "create", "patch" ]
+  - apiGroups: [ "apps" ]
+    resources: [ "deployments" ]
+    verbs: [ "get", "list", "watch", "update", "create", "patch" ]
+  - apiGroups: [ "route.openshift.io" ]
+    resources: [ "routes" ]
+    verbs: [ "get", "list", "watch", "update", "create", "patch" ]
```

Comment:
We cannot grant the operator cluster-wide privileges to create configmaps, secrets, pods, services, persistentvolumeclaims, deployments, routes etc. as this is too permissive. The operator will have to manage these resource only for its own namespace and should be in the namspaced Role only (as I see you have added).

We can grant clusterwide permissions for our own resources as you have done below.

Reply:
That might be an issue with the operator framework then … from what I saw you need to configure this (namedspaced vs non-namespaced) for the whole controller … so we might need to split that up into two applications then.

Reply:
I think you can configure to watch local namepace, and watch resources at the cluster level individually. Many users will be upset by this level of privilege requirements, so we cannot do cluster-wide watching on resources we don't own ourselves.

Reply:
> I think you can configure to watch local namepace, and watch resources at the cluster level for a particular type.

I fail to see that option, do you have a pointer for me?

Reply:
I thought i saw it in the docs somewhere, but now that I'm looking for it I can't find it. Looking at the code, it seems it is not possible through the controller-runtime abstraction but potentially when using informers directly?

Reply:
So instead of re-implementing all the existing code, I would say that we simple split it into two application instances, running with different service accounts. We already can do this with the module flags.

Reply:
Agreed, we should do this short term. However I think we need to fix this longer term, hopefully there will be support in controller-runtime.

Reply:
I split off a second instance, re-using the image, but creating a second service account.

While the service account will be created in the YAML part, the deployment of the operator is being done by the controller-manager. Otherwise the controller-manager would require the permissions to create cluster roles and cluster role bindings, which would somehow seem to contradict the while split.

Reply:
Not required for this PR, but I think we should deploy the operator explicitly through in a iot/operator/deployment-iot-operator.yaml as well. This makes it clearer that you can opt-in the whole iot-thing. We can then in the first instance also create an enmasse-iot operator in operator hub until the global namespace issue is resolved.

I will follow up this in a subsequent PR, so lets just leave this as is for now.